### PR TITLE
add docs for new predicates dsl

### PR DESCRIPTION
**Pull request by ahgittin**
documentation for brooklyn server PR (reference to follow)

### CODE REVIEW

File: guide/blueprints/yaml-reference.md [hunk 1]
```diff
@@ -323,3 +327,145 @@ These are:
...
+* `less-than: <object>`, for strings and primitives, computed using "natural ordering",
+  numeric order for numbers and digit sequences within numbers (`"9" < "10"`),
+  and ASCII-lexicographic comparison elsewhere (`"a" < "b"`);
+  otherwise if both arguments are the same type, or one a subtype of the other, and both comparable,
+  it will use that type's compare method;
+  otherwise if one side side is JSON, it will attempt coercion to the other argument's type;
+  and otherwise it will return false
+* `greater-than: <object>`, as above 
+* `less-than-or-equal-to: <object>` as above
+* `greater-than-or-equal-to: <object>, as above
```
Comment (iuliana) [inline guide/blueprints/yaml-reference.md:359]: missing closing oblique quote

File: guide/blueprints/yaml-reference.md [hunk 2]
```diff
@@ -323,3 +327,145 @@ These are:
...
+To assist with tests on an `Entity` or `Location`, additional keys are supported on
+the `DslPredicate` via `DslEntityPredicate`:
+
+* `target: <expression>`: to specify a value to test, resolved relative to the "context entity"
+  (the one being tested, if appropriate, or otherwise the evaluating entity where the predicate is defined); 
+  this can be a DSL expression such as `$brooklyn:config("key")` 
+  or a keyword, currently `location` or `children`, to evaluate
+  against the location(s) or children of that entity
+* `config: <string>`: indicates that the tests should be applied to the value of config key
+  `<string> on the context entity, location, or policy
```
Comment (iuliana) [inline guide/blueprints/yaml-reference.md:403]: missing closing oblique quote

File: guide/blueprints/yaml-reference.md [hunk 3]
```diff
@@ -323,3 +327,145 @@ These are:
...
+* `sensor: <string>`: indicates that the tests should be applied to the value of sensor
+  `<string> on the context entity
```
Comment (iuliana) [inline guide/blueprints/yaml-reference.md:405]: missing closing oblique quote

**Review by iuliana — APPROVED** [review]

**Comment by ahgittin** [comment]
thanks for fixing!

**Pull request merged**